examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+      "Then, we add the words to a vocabulary with `parse`:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": true,
+     "input": [
+      "dim = 32\n",
+      "vocab = spa.Vocabulary(dimensions=dim)\n",
+      "\n",
+      "words = ['ORANGE', 'APRICOT', 'CHERRY', 'STRAWBERRY', 'APPLE']\n",
+      "\n",
+      "for word in words:\n",
+      "    vocab.parse(word)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Next, we create an autoassociative memory with a corresponding set of stored patterns.\n",
+      "The memory is created within a `nengo.Network` module."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "with nengo.Network('AssociativeMemory', seed=1) as model1:\n",
+      "    assoc_mem = spa.AssociativeMemory(input_vocab=vocab)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "After the network has been created, we can test its functionality by presenting a semantic pointer at the input and comparing it with a semantic pointer at the output after running a simulation. The comparison is done by calculating the dot product. If the similarity between the output vector and the input vector is close to one, we can say that the associative memory successfully retrieved the pattern.\n",
+      "To achieve this task, we need to introduce two additional objects: `nengo.Node` and `nengo.Probe`. The `nengo.Node` object provides input to the associative memory. \n",
+      "`nengo.Probe` is used to gather simulation data, in our case the input being fed to the network (input pattern) and the network output (output pattern) at every simulation time step.\n",
+      "In this example, the input to the network is the semantic pointer `APPLE`:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "with model1:\n",
+      "    input_vector = vocab['APPLE'].v\n",
+      "    input_node = nengo.Node(output=input_vector, label='input')\n",
+      "    \n",
+      "    nengo.Connection(input_node, assoc_mem.input)\n",
+      "\n",
+      "    input_probe = nengo.Probe(input_node)\n",
+      "    output_probe = nengo.Probe(assoc_mem.output, synapse=0.03)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We run the simulation for the 0.2 simulation time:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "sim = nengo.Simulator(model1)\n",
+      "sim.run(0.2)\n",
+      "t = sim.trange()\n",
+      "print('Number of simulation steps: %d' % len(t))"
```

Comment:
I would remove this line as I don't see relevance of the number of simulation steps here.